build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/rewrite/AutobackportFuncTest.groovy

```diff
@@ -0,0 +1,262 @@
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.rewrite;
+
+import org.elasticsearch.gradle.fixtures.AbstractGradleFuncTest;
+
+class AutobackportFuncTest extends AbstractGradleFuncTest {
+
+    def setup() {
+        file("src/main/java/org/elasticsearch/core/List.java") << """
+        package org.elasticsearch.core;
+        public class List {
+            public static <T> java.util.List<T> of(T... entries) {
+                //impl does not matter
+                return new java.util.ArrayList();
+            }
+        }
+        """
+        file("src/main/java/org/elasticsearch/core/Set.java") << """
+        package org.elasticsearch.core;
+        public class Set {
+            public static <T> java.util.Set<T> of(T... entries) {
+                //impl does not matter
+                return new java.util.HashSet();
+            }
+        }
+        """
+    }
+    def "can run rewrite to backport java util methods"() {
+        when:
+        setupRewriteYamlConfig()
+        def sourceFile = file("src/main/java/org/acme/SomeClass.java")
+        sourceFile << """
+package org.acme;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+class SomeClass {
+    public void someMethod() {
+        List myList = List.of("some", "non", "java8", "code");
+        Set mySet = Set.of("some", "non", "java8", "code");
+        Map myMap = Map.of(List.of("some", "non"), Set.of("java8", "code"));
+    }
+}
+"""
+
+        buildFile.text = """
+        plugins {
+          id 'java'
+          id 'elasticsearch.rewrite'
+        }
+        rewrite {
+            rewriteVersion = "7.10.0"
+            activeRecipe("org.elasticsearch.java.backport.ListOfBackport",
+                    "org.elasticsearch.java.backport.MapOfBackport",
+                    "org.elasticsearch.java.backport.SetOfBackport")
+            configFile = rootProject.file("rewrite.yml")
+        }
+        
+        repositories {
+            mavenCentral()
+            maven { url 'https://jitpack.io' }
+        }
+        
+        dependencies {
+            rewrite "org.openrewrite:rewrite-java-11"
+            rewrite "com.github.breskeby:java-recipes:0dde6854d5"
+        }
+        """
+
+        then:
+        gradleRunner("rewrite").build()
+
+        sourceFile.text == """
+package org.acme;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+class SomeClass {
+    public void someMethod() {
+        List myList = org.elasticsearch.core.List.of("some", "non", "java8", "code");
+        Set mySet = org.elasticsearch.core.Set.of("some", "non", "java8", "code");
+        Map myMap = org.elasticsearch.core.Map.of(org.elasticsearch.core.List.of("some", "non"), org.elasticsearch.core.Set.of("java8", "code"));
+    }
+}
+"""
+    }
+
+    def "converts new full qualified usage of elastic util methods where possible"() {
```

Comment:
Given this test name, I expected to see e.g. `java.util.List.of()` in the sample program, but none of the usages are fully qualified?

Reply:
If I understand correctly, the purpose of this test is to assert that `org.elasticsearch.core.List.of()` becomes `List.of()` and the appropriate import statement is added.

Reply:
correct